onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -2528,4 +2528,32 @@ namespace OperatorHelper
         return { EdgeShapes({}) };
     }
 
+    std::vector<EdgeShapes> EmbedLayerNormalizationHelper::GetOutputShapes(const MLShapeInferenceContext & shapeInfo) const
+    {
+        ML_CHECK_VALID_ARGUMENT(shapeInfo.GetInputCount() >= 3);
+
+        // input_ids and word_embedding are 2D tensors
+        ML_CHECK_VALID_ARGUMENT(shapeInfo.GetInputTensorShape(0).size() == 2);
+        ML_CHECK_VALID_ARGUMENT(shapeInfo.GetInputTensorShape(2).size() == 2);
+
```

Comment:
[minor] If we move the CHECK_VALID after line 2540, we can read `inputIdsShape` and `wordEmbeddingShape`. Though, I don't think `GetInputTensorShape` very heavy (probably just an std vector allocation), and I can understand if you want all the `CHECK` statements up front.